strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BaseArtifactController.java

```diff
@@ -96,10 +97,12 @@ protected boolean probeForDirectoryListing(Repository repository,
         String dir = repository.getBasedir() + File.separator + filePath;
 
         File file = new File(dir);
-
-        // Do not allow .index and .trash directories (or any other directory starting with ".") to be browseable.
+        
+        // For Maven 2 Repositories only allow .index to be browsable directly by admins
+        // For others do not allow .index and .trash directories (or any other directory starting with ".") to be browseable.
         // NB: Files will still be downloadable.
-        if (!file.isHidden() && !path.startsWith(".") && !path.contains("/."))
+        if ((repository.getLayout().equals(RepositoryLayoutEnum.MAVEN_2.getLayout()) && path.startsWith(".index"))
```

Comment:
Would you mind extracting a template method here and override it in the `org.carlspring.strongbox.controllers.maven.MavenArtifactController` ? That way we'll leave the `org.carlspring.strongbox.controllers.BaseArtifactController` more generic without any repository layout specific information.